spec/features/trainees/degrees/editing_degree_spec.rb

```diff
@@ -93,9 +95,13 @@ def trainee
 
   def uk_trainee
     @uk_trainee ||= create(:degree, :uk_degree_type).trainee
+    @uk_trainee.provider = current_user.provider
```

Comment:
I think you can just pass the provider in the create `create(:degree, :uk_degree_type, provider: current_user.provider).trainee`

Reply:
I did try this, and it was having issues remembering that I'd set the provider. I ended up getting it to work with this, but I'm not sure why it wasn't working - I think it's something to do with the fact the trainee is created by association in the degree factory, which means it doesn't take custom parameters like that.

Reply:
Sorted!